src/main/java/com/internet/shop/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,185 @@
+            preparedStatement.setLong(1, userId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                Long orderId = resultSet.getLong("order_id");
+                orderList.add(new Order(orderId, userId));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to get orders of user with id" + userId, e);
+        }
+        for (Order order : orderList) {
+            List<Product> productList = getOrdersProducts(order.getOrderId());
+            order.setProducts(productList);
+        }
+        return orderList;
+    }
+
+    @Override
+    public Optional<Order> getByUserId(Long id) {
+        return Optional.empty();
+    }
+
+    private void addDataToOrdersProductTable(Long orderId, Long productId) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection
+                        .prepareStatement("INSERT INTO "
+                             + "orders_products (order_id, product_id) VALUES(?,?)")) {
+            statement.setLong(1, orderId);
+            statement.setLong(2, productId);
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add data to Orders_Product", e);
+        }
+    }
+
+    private List<Product> getOrdersProducts(long orderId) {
+        List<Product> products = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection(); PreparedStatement statement =
+                connection.prepareStatement("SELECT * FROM products p JOIN "
+                        + "orders_products op ON op.product_id = p.product_id "
+                        + "WHERE order_id = ? AND is_deleted = FALSE")) {
+            statement.setLong(1, orderId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                products.add(new Product(resultSet.getString("name"),
+                        resultSet.getDouble("price"),
+                        resultSet.getLong("product_id"),
+                        resultSet.getBoolean("deleted")));
+            }
+
+            return products;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get Order Products", e);
+        }
+    }
+
+    public boolean deleteOrderProducts(Long orderId) {
+        String deleteQuery = "DELETE FROM orders_products WHERE order_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(deleteQuery)) {
+            statement.setLong(1, orderId);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete Order with id: " + orderId, e);
+        }
+    }
```

Comment:
I don't think we need to do this. It would be good to keep the history of products in orders in the database even if we set 'is_deleted = true' for the order